src/main/java/net/dv8tion/jda/core/requests/RestAction.java

```diff
@@ -615,18 +652,44 @@ public T completeAfter(long delay, TimeUnit unit)
      */
     public ScheduledFuture<?> queueAfter(long delay, TimeUnit unit, Consumer<T> success, Consumer<Throwable> failure, ScheduledExecutorService executor)
     {
-        Args.notNull(executor, "Scheduler");
-        Args.notNull(unit, "TimeUnit");
+        Checks.notNull(executor, "Scheduler");
+        Checks.notNull(unit, "TimeUnit");
         return executor.schedule(() -> queue(success, failure), delay, unit);
     }
 
-    protected void finalizeData() { }
+    private final void finalizeAction() {
+        this.data = finalizeData();
+        this.route = this.finalizeRoute();
+        this.headers = this.finalizeHeaders();
+    }
+
+    protected RequestBody finalizeData() { return data; }
+    protected CompiledRoute finalizeRoute() { return route; }
+    protected CaseInsensitiveMap<String, String> finalizeHeaders() { return headers; }
+
+    protected Route.CompiledRoute getRoute()
+    {
+        return this.route;
+    }
+
+    protected RequestBody getData()
+    {
+        return this.data;
+    }
+
+    protected CaseInsensitiveMap<String, String> getHeaders()
+    {
+        return this.headers;
+    }
```

Comment:
Good catch, didn't think about this. But I think this is also problematic for the fields. Fixed it in the next commit.